src/Futhark/Fmt/Monad.hs

```diff
@@ -314,143 +311,140 @@ hardline = do
       pure P.line
 
 -- | A line or a space depending on layout.
-line :: FmtM Fmt
+line :: Fmt
 line = space <|> hardline
 
 -- | A comment.
-comment :: T.Text -> FmtM Fmt
+comment :: T.Text -> Fmt
 comment c = do
   modify (\s -> s {lastOutput = Just Line})
   pure $ P.pretty c <> P.line
 
 -- in order to handle trailing comments its VERY important to
 -- evaluate the seperator after each element in the list.
-sep :: (Format a, Format b) => a -> [b] -> FmtM Fmt
+sep :: Fmt -> [Fmt] -> Fmt
```

Comment:
We need to know the location to add comments inside, but it maybe prependComments should be exported and used in Printer.hs. This change currently ruin comments in some cases.